activitysim/abm/models/vehicle_allocation.py

```diff
@@ -0,0 +1,219 @@
+    vehicles_wide = vehicles.to_frame().pivot_table(
+        index='household_id', columns='vehicle_num',
+        values='vehicle_type', aggfunc=lambda x: ''.join(x))
+
+    alts_from_spec = model_spec.columns
+    # renaming vehicle numbers to alternative names in spec
+    vehicle_alt_columns_dict = {}
+    for veh_num in range(1, len(alts_from_spec)):
+        vehicle_alt_columns_dict[veh_num] = alts_from_spec[veh_num-1]
+    vehicles_wide.rename(columns=vehicle_alt_columns_dict, inplace=True)
+
+    # if the number of vehicles is less than the alternatives, fill with NA
+    # e.g. all households only have 1 or 2 vehicles because of small sample size,
+    #   still need columns for alternatives 3 and 4
+    for veh_num, col_name in vehicle_alt_columns_dict.items():
+        if col_name not in vehicles_wide.columns:
+            vehicles_wide[col_name] = ''
+
+    # last entry in spec is the non-hh-veh option
+    assert alts_from_spec[-1] == 'non_hh_veh', "Last option in spec needs to be non_hh_veh"
+    vehicles_wide[alts_from_spec[-1]] = ''
+
+    # merging vehicle alternatives to choosers
+    choosers = tours_merged.to_frame().reset_index()
+    choosers = pd.merge(choosers, vehicles_wide, how='left', on='household_id')
+    choosers.set_index('tour_id', inplace=True)
+
+    # ----- setup skim keys
+    skims = get_skim_dict(network_los, choosers)
+    locals_dict.update(skims)
+
+    # ------ preprocessor
+    preprocessor_settings = model_settings.get('preprocessor', None)
+    if preprocessor_settings:
+        expressions.assign_columns(
+            df=choosers,
+            model_settings=preprocessor_settings,
+            locals_dict=locals_dict,
+            trace_label=trace_label)
+
+    logger.info("Running %s with %d tours", trace_label, len(choosers))
+
+    if estimator:
+        estimator.write_model_settings(model_settings, model_settings_file_name)
+        estimator.write_spec(model_settings)
+        estimator.write_coefficients(coefficients_df, model_settings)
+        estimator.write_choosers(choosers)
+
+    tours = tours.to_frame()
+
+    # ------ running for each occupancy level selected
+    tours_veh_occup_cols = []
+    for occup in model_settings.get('OCCUPANCY_LEVELS', [1]):
+        logger.info("Running for occupancy = %d", occup)
+        # setting occup for access in spec expressions
+        locals_dict.update({'occup': occup})
+
+        choices = simulate.simple_simulate(
+            choosers=choosers,
+            spec=model_spec,
+            nest_spec=nest_spec,
+            skims=skims,
+            locals_d=locals_dict,
+            chunk_size=chunk_size,
+            trace_label=trace_label,
+            trace_choice_name='vehicle_allocation',
+            estimator=estimator)
+
+        # matching alt names to choices
+        choices = choices.map(dict(list(zip(list(range(len(alts_from_spec))), alts_from_spec)))).to_frame()
```

Comment:
suggest the more legible:
```python
choices.map(dict(enumerate(alts_from_spec))).to_frame()
```

Reply:
ah, much cleaner, thanks!  Also changed where appropriate in vehicle_type_choice.py.